ethdb/privateapi/ethbackend.go

```diff
@@ -221,19 +213,6 @@ func (s *EthBackendServer) EngineExecutePayloadV1(ctx context.Context, req *type
 		eip1559 = true
 	}
 
-	// Decode transactions
-	reader := bytes.NewReader(nil)
-	stream := rlp.NewStream(reader, 0)
-	transactions := make([]types.Transaction, len(req.Transactions))
-
-	for i, encodedTransaction := range req.Transactions {
-		reader.Reset(encodedTransaction)
-		stream.Reset(reader, 0)
-		if transactions[i], err = types.DecodeTransaction(stream); err != nil {
```

Comment:
We shouldn't get rid of transaction decoding­­ – otherwise we are not checking that the provided transaction RLP is valid. Moreover, we'll need transactions themselves to payload(i.e. block) execution.

Reply:
we can check if the transactions are valid by comparing the hash of the payload with the hash of the header + TransactionRoot. note also that everything coming from the beacon chain is assumed to be valid, if something makes no sense later on, we return status Invalid